contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/statsbeat_metrics/statsbeat.py

```diff
@@ -80,78 +85,140 @@ def _get_attach_properties():
     return properties
 
 
+def _get_attach_properties():
+    properties = _get_common_properties()
+    properties.insert(1, LabelKey("rpid", 'unique id of rp'))
+    return properties
+
+
+def _get_network_properties():
+    properties = _get_common_properties()
+    return properties
+
+
+def _get_success_count_value():
+    with _requests_lock:
+        interval_count = _requests_map.get('success', 0) \
+                    - _requests_map.get('last_success', 0)
+        _requests_map['last_success'] = _requests_map.get('success', 0)
+        return interval_count
+
+
 class _StatsbeatMetrics:
 
     def __init__(self, instrumentation_key):
         self._instrumentation_key = instrumentation_key
-        self.vm_data = {}
-        self.vm_retry = True
+        self._vm_data = {}
+        self._vm_retry = True
+        self._rp = _RP_NAMES[3]
+        self._os_type = platform.system()
         # Attach metrics - metrics related to rp (resource provider)
         self._attach_metric = LongGauge(
             _ATTACH_METRIC_NAME,
             'Statsbeat metric related to rp integrations',
             'count',
             _get_attach_properties(),
         )
+        # Keep track of how many iterations until long export
+        self._long_threshold_count = 0
+        # Network metrics - metrics related to request calls to Breeze
+        self._network_metrics = {}
+        # Map of gauge function -> metric
+        # Gauge function is the callback used to populate the metric value
+        self._network_metrics[_get_success_count_value] = DerivedLongGauge(
+            _REQ_SUC_COUNT_NAME,
+            'Request success count',
+            'count',
+            _get_network_properties(),
+        )
 
+    # Metrics that are sent on application start
     def get_initial_metrics(self):
         stats_metrics = []
         if self._attach_metric:
-            attach_metric = self._get_attach_metric(self._attach_metric)
+            attach_metric = self._get_attach_metric()
             if attach_metric:
                 stats_metrics.append(attach_metric)
         return stats_metrics
 
+    # Metrics sent every statsbeat interval
     def get_metrics(self):
-        stats_metrics = self.get_initial_metrics()
-
-        return stats_metrics
-
-    def _get_attach_metric(self, metric):
+        metrics = []
+        try:
+            # Initial metrics use the long export interval
+            # Only export once long count hits threshold
+            self._long_threshold_count = self._long_threshold_count + 1
+            if self._long_threshold_count >= _STATS_LONG_INTERVAL_THRESHOLD:
```

Comment:
how does this threshold scale when shorter and long intervals get changed later? can you use the explicit intervals? what about multi-threading env?

Reply:
What do you mean if the intervals get changed later? The intervals are defined once during start of the application and the threshold is calculated based off of those at that time.

> what about multi-threading env?

I'll add a locking mechanism to `_StatsbeatMetric`.

Reply:
not really.. we might increate the attach internal.. or any other statsbeat metrics.  interval can be adjusted later based ont he data we collect.. it will offer some insights whether or not it's a good idea to update it.

Reply:
Right, for now there's only two different interval types. If more are added, more thresholds can be created. It's not worth to define separate exporter threads for each type of interval.